internal/provider/resource_copy.go

```diff
@@ -0,0 +1,178 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/chainguard-dev/terraform-provider-oci/pkg/validators"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/sigstore/cosign/v2/cmd/cosign/cli/copy"
+	"github.com/sigstore/cosign/v2/cmd/cosign/cli/options"
+)
+
+var (
+	_ resource.Resource                = &CopyResource{}
+	_ resource.ResourceWithImportState = &CopyResource{}
+)
+
+func NewCopyResource() resource.Resource {
+	return &CopyResource{}
+}
+
+type CopyResource struct {
+}
+
+type CopyResourceModel struct {
+	Id              types.String `tfsdk:"id"`
+	Source          types.String `tfsdk:"source"`
+	DestinationRepo types.String `tfsdk:"destination_repo"`
+
+	CopiedRef types.String `tfsdk:"copied_ref"`
+}
+
+func (r *CopyResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_copy"
+}
+
+func (r *CopyResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This copies the provided image digest cosign copy.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The immutable digest this resource copies, along with its signatures, etc.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"source": schema.StringAttribute{
+				MarkdownDescription: "The digest of the container image to copy.",
+				Optional:            false,
+				Required:            true,
+				Validators:          []validator.String{validators.DigestValidator{}},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"destination_repo": schema.StringAttribute{
```

Comment:
I don't love how verbose this is.  Can we make it `destination` and let the type qualify it?